micrometer-core/src/main/java/io/micrometer/core/instrument/binder/http/AbstractDefaultHttpClientRequestObservationConvention.java

```diff
@@ -0,0 +1,140 @@
+
+    private static final KeyValue EXCEPTION_NONE = KeyValue.of(CommonLowCardinalityKeys.EXCEPTION, KeyValue.NONE_VALUE);
+
+    private static final KeyValue HTTP_URL_NONE = KeyValue.of(CommonHighCardinalityKeys.HTTP_URL, KeyValue.NONE_VALUE);
+
+    private static final KeyValue USER_AGENT_NONE = KeyValue.of(CommonHighCardinalityKeys.USER_AGENT_ORIGINAL,
+            KeyValue.NONE_VALUE);
+
+    protected String getContextualName(String lowercaseMethod) {
+        return "http " + lowercaseMethod;
+    }
+
+    protected KeyValues getLowCardinalityKeyValues(@Nullable URI uri, @Nullable Throwable throwable,
+            @Nullable String methodName, @Nullable Integer statusCode, @Nullable String uriPathPattern) {
+        return KeyValues.of(clientName(uri), exception(throwable), method(methodName), outcome(statusCode),
+                status(statusCode), uri(uriPathPattern));
+    }
+
+    private KeyValue uri(@Nullable String uriTemplate) {
+        if (uriTemplate != null) {
+            return KeyValue.of(CommonLowCardinalityKeys.URI, extractPath(uriTemplate));
+        }
+        return URI_NONE;
+    }
+
+    private static String extractPath(String uriTemplate) {
+        String path = PATTERN_BEFORE_PATH.matcher(uriTemplate).replaceFirst("");
+        return (path.startsWith("/") ? path : "/" + path);
+    }
+
+    private KeyValue method(@Nullable String methodName) {
+        if (methodName != null) {
+            return KeyValue.of(CommonLowCardinalityKeys.METHOD, methodName);
+        }
+        else {
+            return METHOD_NONE;
+        }
+    }
+
+    private KeyValue status(@Nullable Integer statusCode) {
+        if (statusCode == null) {
+            return STATUS_CLIENT_ERROR;
+        }
+        return KeyValue.of(CommonLowCardinalityKeys.STATUS, String.valueOf(statusCode));
+    }
+
+    private KeyValue clientName(@Nullable URI uri) {
+        if (uri != null && uri.getHost() != null) {
+            return KeyValue.of(ClientLowCardinalityKeys.CLIENT_NAME, uri.getHost());
+        }
+        return CLIENT_NAME_NONE;
+    }
+
+    private KeyValue exception(@Nullable Throwable error) {
+        if (error != null) {
+            String simpleName = error.getClass().getSimpleName();
+            return KeyValue.of(CommonLowCardinalityKeys.EXCEPTION,
+                    StringUtils.isNotBlank(simpleName) ? simpleName : error.getClass().getName());
+        }
+        return EXCEPTION_NONE;
+    }
+
+    private KeyValue outcome(@Nullable Integer statusCode) {
+        if (statusCode != null) {
+            return HttpOutcome.forStatus(false, statusCode);
+        }
+        return HttpOutcome.forStatus(false, statusCode);
+    }
+
+    protected KeyValues getHighCardinalityKeyValues(@Nullable URI uri, @Nullable String userAgent) {
+        // Make sure that KeyValues entries are already sorted by name for better
+        // performance
+        return KeyValues.of(requestUri(uri), userAgent(userAgent));
```

Comment:
I think user-agent is a great idea and we should do it in Spring Framework too.
/cc @bclozel